### PR TITLE
swift-module-digester: diagnose operator *fix notation changes.

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test and merge

File: test/api-digester/Inputs/cake2.swift [hunk 1]
```diff
@@ -141,3 +141,5 @@ public class EscapingFunctionType {
   public func removedEscaping(_ a: ()->()) {}
   public func addedEscaping(_ a: @escaping ()->()) {}
 }
+
+prefix operator ..*..
```
Comment (jrose-apple) [inline test/api-digester/Inputs/cake2.swift]: Hm. What happens if we add the prefix operator but don't remove the infix operator? Is that correctly recorded as a source-compatible addition?

Comment (nkcsgexi) [reply test/api-digester/Inputs/cake2.swift]: no, we don't treat any additional decls as source/ABI breaking except stored property in non-resilient types and added requirements. Are added operators always source breaking?

Comment (jrose-apple) [reply test/api-digester/Inputs/cake2.swift]: Sorry, they're not source-breaking. I guess I just want a test that it's not counted as source-breaking by accident.

Comment (nkcsgexi) [reply test/api-digester/Inputs/cake2.swift]: ah, no, it won't. Two infix operators will match with each other for comparison and the added prefix operator will be a source-compatible addition.

**Pull request merged**